modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java

```diff
@@ -393,4 +400,46 @@ public Tomcat getTomcat() {
 
 		return tomcat;
 	}
+
+    /**
+     * Request access token from the given token generation endpoint
+     *
+     * @param consumerKey    consumer key of the application
+     * @param consumerSecret consumer secret of the application
+     * @param backendUrl     token generation API endpoint
+     * @return token
+     * @throws Exception if something went wrong when requesting token
+     */
+    public static String requestAccessToken(String consumerKey, String consumerSecret,
+                                            String backendUrl, String username, String password) throws Exception {
+        ArrayList<NameValuePair> postParameters;
+        HttpClient client = new DefaultHttpClient();
+        HttpPost httpPost = new HttpPost(backendUrl);
+        //generate post request
+        httpPost.setHeader("Authorization", "Basic " + getBase64EncodedString(consumerKey, consumerSecret));
+        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded");
+        postParameters = new ArrayList<NameValuePair>();
+        postParameters.add(new BasicNameValuePair("username", username));
+        postParameters.add(new BasicNameValuePair("password", password));
+        postParameters.add(new BasicNameValuePair("scope", SCOPE_PRODUCTION));
+        postParameters.add(new BasicNameValuePair("grant_type", GRANT_TYPE_PASSWORD));
+        httpPost.setEntity(new UrlEncodedFormEntity(postParameters));
+        HttpResponse response = client.execute(httpPost);
+        String responseString = EntityUtils.toString(response.getEntity(), "UTF-8");
+        //Get access token from the response
+        JSONParser parser = new JSONParser();
+        JSONObject json = (JSONObject) parser.parse(responseString);
+        return json.get("access_token").toString();
+    }
+
+    /**
+     * Get base64 encoded string of consumer key and secret
+     *
+     * @param consumerKey    consumer key of the application
+     * @param consumerSecret consumer secret of the application
+     * @return base 64 encoded string
+     */
+    public static String getBase64EncodedString(String consumerKey, String consumerSecret) {
```

Comment:
Following the above argument, this method should go to some OAuthTestUtil class.